From: Miroslav Benes <mbenes@suse.cz>
Date: Thu, 17 Mar 2016 13:44:55 +0100
Subject: kgr: fix reversion of a patch already reverted by a replace_all patch
Patch-mainline: not yet, kgraft
References: fate#313296

When a replace_all patch is applied all previous patches are reverted
automatically. Among others the patches are removed from the list of
applied patches and their modules' references are decreased.

Unfortunately there is a bug there. If one tries to revert the reverted
patch manually via sysfs interface he gets general protection fault. The
problem is in kgr_modify_kernel() and its check for reversion. It is
based on list_empty() call but we delete the patch from the list only
with list_del() and not with list_del_init().

Signed-off-by: Miroslav Benes <mbenes@suse.cz>
Reviewed-by: Jiri Kosina <jkosina@suse.cz>
Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 kernel/kgraft.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/kernel/kgraft.c
+++ b/kernel/kgraft.c
@@ -170,7 +170,7 @@ static void kgr_remove_patches_fast(void
 	struct kgr_patch *p, *tmp;
 
 	list_for_each_entry_safe(p, tmp, &kgr_patches, list) {
-		list_del(&p->list);
+		list_del_init(&p->list);
 		module_put(p->owner);
 	}
 }
